Mac OS X 10.6 and earlier: Undefined symbols: _strnlen

libgit2 0.20.0 builds fine on Mac OS X 10.7, 10.8, and 10.9, but fails on 10.6 (and presumably earlier) with this error:

```
Undefined symbols:
  "_strnlen", referenced from:
      _git_index_read in index.c.o
      _git_index_read in index.c.o
      _git_index_read in index.c.o
      _git_index_read in index.c.o
ld: symbol(s) not found
collect2: ld returned 1 exit status
```

Full log: https://build.macports.org/builders/buildports-snowleopard-x86_64/builds/23352/steps/compile/logs/stdio

--- a/src/posix.h
+++ b/src/posix.h
@@ -89,13 +89,17 @@ extern struct tm * p_gmtime_r (const time_t *timer, struct tm *result);
 #	include "unix/posix.h"
 #endif
 
-#if defined(__MINGW32__) || defined(__sun)
+#if defined(__MINGW32__) || defined(__sun) || defined(__APPLE__)
+#   define NO_STRNLEN
+#endif
+
+#ifdef NO_STRNLEN
 GIT_INLINE(size_t) p_strnlen(const char *s, size_t maxlen) {
 	const char *end = memchr(s, 0, maxlen);
 	return end ? (size_t)(end - s) : maxlen;
 }
 #else
-# define p_strnlen strnlen
+#   define p_strnlen strnlen
 #endif
 
 #ifdef NO_READDIR_R

